### PR TITLE
Gallery page

### LINKED ISSUE #12: Create gallery page

### PR DESCRIPTION
**Pull request by arianazhu**
Added gallery page using the test JSON file to display images with captions.

### CODE REVIEW

File: src/main/webapp/gallery.css [hunk 1]
```diff
@@ -10,8 +10,20 @@ body {
     font-family: sans-serif;
 }
 
+#refresh-button {
+    margin-left: 7px;
```
Comment (fgoyal) [inline src/main/webapp/gallery.css]: Super nitpick, but the indentation on this doesn't match the others.

File: src/main/webapp/gallery.css [hunk 2]
```diff
@@ -22,14 +34,36 @@ body {
...
+  margin: 0 !important;
+}
+
+/* The overlay effect - lays on top of the container and over the image */
+.overlay {
+  position: absolute;
+  bottom: 0;
+  background: rgb(0, 0, 0);
+  background: rgba(0, 0, 0, 0.5); /* Black see-through */
+  color: #f1f1f1;
```
Comment (fgoyal) [inline src/main/webapp/gallery.css]: Future reference note: need to change this to variables for a uniform color scheme. I need to see if we can use the color scheme in style.css across css files.

File: src/main/webapp/gallery.html [hunk 3]
```diff
@@ -3,17 +3,18 @@
...
         <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
+
     </head>
 
     <body>
         <!-- Navigation -->
         <nav class="navbar navbar-expand-lg navbar-dark bg-dark static-top">
             <div class="container">
-                <a class="navbar-brand" href="#">Pieced</a>
+                <a class="navbar-brand" href="index.html">Pieced</a>
```
Comment (fgoyal) [inline src/main/webapp/gallery.html]: index.html is unnecessary here; we can just use "/"

File: src/main/webapp/index.css [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  background-color: var(--link-hover-color);
+}
+
+ul {
+  background-color: var(--list-color);
+  list-style-type: none;
+  margin: 0;
+  overflow: hidden;
+  padding: 0;
+}
```
Comment (fgoyal) [inline src/main/webapp/index.css:62]: Might need to move these to style.css so we have the same styles across pages. No change needed now though.

File: src/main/webapp/gallery.js [hunk 5]
```diff
@@ -1,13 +1,34 @@
...
+    percentPosition: true,
+    columnWidth: '.grid-sizer',
+    horizontalOrder: true
 });
-// Layout Masonry after each image finishes loading
-$grid.imagesLoaded().progress( function() {
-  $grid.masonry();
-});  
+
+fetchSpeciesData();
```
Comment (fgoyal) [inline src/main/webapp/gallery.js]: Since you have body onload call this function, I don't think you need to call it here?

**Review by fgoyal — APPROVED** [review]

File: CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# These owners will be the default owners for everything in
+# the repo. Unless a later match takes precedence.
+*	@taruss2011, @dnewmanmax
```
Comment (maxhchen) [inline CODEOWNERS:3]: Didn't know we needed to add this, was it mentioned somewhere?

File: src/main/webapp/gallery.html [hunk 7]
```diff
@@ -3,17 +3,18 @@
...
-        <title>Gallery</title>
+        <title>Pieced - Gallery</title>
 
         <link rel="stylesheet" href="gallery.css">
         <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
+
     </head>
 
     <body>
         <!-- Navigation -->
```
Comment (maxhchen) [inline src/main/webapp/gallery.html]: I think we could replace this with the navbar that Fiza was working on as a next step

File: src/main/webapp/gallery.js [hunk 8]
```diff
@@ -1,13 +1,34 @@
...
+                // add and lay out newly appended items
+                .masonry('appended', $html);
+        }
+
+        // Layout Masonry after each image finishes loading
+        $grid.imagesLoaded().progress( function() {
+            $grid.masonry('layout');
+        });
+    });
+}
```
Comment (maxhchen) [inline src/main/webapp/gallery.js]: nice!

**Review by maxhchen — APPROVED** [review]

File: src/main/webapp/gallery.js (hunk 5 above)
Comment (arianazhu) [reply src/main/webapp/gallery.js]: I think I took it out from the onload call, so it gets called here instead

File: CODEOWNERS (hunk 6 above)
Comment (arianazhu) [reply CODEOWNERS:3]: I'm not sure, but I just merged this from master

File: src/main/webapp/gallery.html (hunk 7 above)
Comment (arianazhu) [reply src/main/webapp/gallery.html]: Yep I'll change that once we merge the navbar!

**Pull request merged**